analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/SymbolicValueAssertions.cs

```diff
@@ -0,0 +1,164 @@
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("The {context:symbolicValue} is not present and can not have constraint {0}.", constraint)
+                .Then
+                .Given(() => Subject.AllConstraints.ToList())
+                .ForCondition(allConstraints => allConstraints.Count != 0)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has no constraints.", constraint)
+                .Then
+                .ForCondition(allConstraints => allConstraints.Contains(constraint))
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has {1} constraints.",
+                    _ => constraint,
+                    x => x.OrderBy(x => x.ToString()));
+            var matchedConstrainted = Subject.AllConstraints.First(x => x == constraint);
+            return new(this, matchedConstrainted);
+        }
+
+        public AndWhichConstraint<SymbolicValueAssertions, SymbolicConstraint> HaveOnlyConstraint(SymbolicConstraint constraint, string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject != null)
+                .FailWith("The {context:symbolicValue} is not present and can not have constraint {0}.", constraint)
+                .Then
+                .Given(() => Subject.AllConstraints.ToList())
+                .ForCondition(allConstraints => allConstraints.Count > 0)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has no constraints.", constraint)
+                .Then
+                .ForCondition(allConstraints => allConstraints.Count == 1)
+                .FailWith("Expected {context:symbolicValue} to have only constraint {0}{reason}, but symbolicValue has {1} constraints.",
+                    _ => constraint,
+                    x => x.OrderBy(x => x.ToString()))
+                .Then
+                .ForCondition(allConstraints => allConstraints[0] == constraint)
+                .FailWith("Expected {context:symbolicValue} to have constraint {0}{reason}, but symbolicValue has {1} constraint.",
+                    _ => constraint,
+                    allConstraints => allConstraints[0]);
+            var matchedConstrainted = Subject.AllConstraints.First(x => x == constraint);
+            return new(this, matchedConstrainted);
+        }
+
+        public AndConstraint<SymbolicValueAssertions> HaveNoConstraints(string because = "", params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject == null || Subject.AllConstraints.All(x => x == null))
```

Comment:
`null` value cannot be stored in constraint collection
```suggestion
                .ForCondition(Subject == null || Subject.AllConstraints..Count() == 0)
```